### DISCUSSION TITLE
Set parent page when bulk importing pages

### DISCUSSION THREAD

**Discussion by sewid** [statamic/cms#6224] [Q&A] [answered]
Hi there,

I have a multisite setup with multiple locales and try to create about 300 pages by code. Everything works fine, except building the tree. I cannot set the parent for pages. Has anybody a working code example?

The error I always get is:

```
Page [cb5c190d-b1b6-42f7-b8e1-e46e7a81b45a] does not exist in this structure
```

But the mentioned ID exists in my collection in the desired locale?!

I already tried these examples:
https://github.com/statamic/ideas/issues/739 and https://statamic.com/forum/4946-set-parent-entry-programatically

Thanks in advance!
Sebastian

**Comment by sewid** [comment, marked as answer]
I've got it working. I have no idea, what went wrong. I deleted my complete code and used this code again:

https://github.com/statamic/ideas/issues/739

At the end, there is an `$entry->save();` missing, but the rest is working.

**Reply by ryanmitchell** [reply]
Glad i could help :)
